### PR TITLE
[sival,pwrmgr] Create dedicated test for wakeup 5

### PR DESCRIPTION
**Pull request by matutem**
Create test for wakeup reason 5, expected to fail due to #20798 from sw/device/tests/pwrmgr_random_sleep_all_wake_ups.c.

Skip wakeup reason 5 in pwrmgr_random_sleep_all_wake_ups, with a TODO referring to #20798.

### CODE REVIEW

**Comment by matutem** [comment]
When this is merged we'll update #20798 repro instructions.

File: sw/device/tests/pwrmgr_sleep_wake_5_bug_test.c [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+/**
+ * Clean up pwrmgr wakeup reason register for the next round.
+ */
+static void delay_n_clear(uint32_t delay_in_us) {
+  busy_spin_micros(delay_in_us);
+  CHECK_DIF_OK(dif_pwrmgr_wakeup_reason_clear(&pwrmgr));
+}
+
+uint32_t get_wakeup_unit(uint32_t count) { return count / 2; }
```
Comment (timothytrippel) [inline sw/device/tests/pwrmgr_sleep_wake_5_bug_test.c:45]: static? same below?

Comment (timothytrippel) [reply sw/device/tests/pwrmgr_sleep_wake_5_bug_test.c:45]: or are these used outside this translation unit?

**Review by timothytrippel — APPROVED** [review]

File: sw/device/tests/pwrmgr_sleep_wake_5_bug_test.c (hunk 1 above)
Comment (matutem) [reply sw/device/tests/pwrmgr_sleep_wake_5_bug_test.c]: Fixed, thanks. Also fixed a few similar cases in related tests.

**Pull request merged**